pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/internal/templates/controllers/controller.go

```diff
@@ -131,6 +139,43 @@ func (r *{{ .Resource.Kind }}Reconciler) Reconcile(ctx context.Context, req ctrl
 		return ctrl.Result{}, err
 	}
 
+	// Let's add a finalizer. Then, we can define some operations which should
+	// occurs before the custom resource to be deleted.
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/finalizers/
+	// NOTE: You should not use finalizer to delete the resources that are
+	// created in this reconciliation and have the ownerRef set by ctrl.SetControllerReference
+	// because these will get deleted via k8s api
+	if !controllerutil.ContainsFinalizer({{ lower .Resource.Kind }}, {{ lower .Resource.Kind }}Finalizer) {
+		controllerutil.AddFinalizer({{ lower .Resource.Kind }}, {{ lower .Resource.Kind }}Finalizer)
```

Comment:
```suggestion
                 log.Info("Adding Finalizer for {{ Resource.Kind }}")
		controllerutil.AddFinalizer({{ lower .Resource.Kind }}, {{ lower .Resource.Kind }}Finalizer)
```
Let's add some logs so that we can check what it is doing and we are able to troubleshooting